src/meta-srv/src/election/mysql.rs

```diff
@@ -0,0 +1,836 @@
+                .await?
+                .unwrap_or_default();
+
+            ensure!(
+                prev_expire_time > current_time,
+                UnexpectedSnafu {
+                    violated: format!(
+                        "Candidate lease expired at {:?} (current time: {:?}), key: {:?}",
+                        prev_expire_time,
+                        current_time,
+                        String::from_utf8_lossy(&key.into_bytes())
+                    ),
+                }
+            );
+
+            // Safety: origin is Some since we are using `get_value_with_lease` with `true`.
+            let origin = origin.unwrap();
+            self.update_value_with_lease(&key, &origin, &node_info, &mut executor)
+                .await?;
+            std::mem::drop(executor);
+        }
+    }
+
+    async fn all_candidates(&self) -> Result<Vec<MetasrvNodeInfo>> {
+        let key_prefix = self.candidate_root();
+        let client = self.client.lock().await;
+        let mut executor = Executor::Default(client);
+        let (mut candidates, current) = self
+            .get_value_with_lease_by_prefix(&key_prefix, &mut executor)
+            .await?;
+        // Remove expired candidates
+        candidates.retain(|c| c.1 > current);
+        let mut valid_candidates = Vec::with_capacity(candidates.len());
+        for (c, _) in candidates {
+            let node_info: MetasrvNodeInfo =
+                serde_json::from_str(&c).with_context(|_| DeserializeFromJsonSnafu {
+                    input: format!("{:?}", c),
+                })?;
+            valid_candidates.push(node_info);
+        }
+        Ok(valid_candidates)
+    }
+
+    async fn campaign(&self) -> Result<()> {
+        let mut keep_alive_interval =
+            tokio::time::interval(Duration::from_secs(META_KEEP_ALIVE_INTERVAL_SECS));
+        keep_alive_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        loop {
+            let _ = self.do_campaign(&mut keep_alive_interval).await;
+        }
+    }
+
+    async fn leader(&self) -> Result<Self::Leader> {
+        if self.is_leader.load(Ordering::Relaxed) {
+            Ok(self.leader_value.as_bytes().into())
+        } else {
+            let key = self.election_key();
+
+            let client = self.client.lock().await;
+            let mut executor = Executor::Default(client);
+            if let Some((leader, expire_time, current, _)) = self
+                .get_value_with_lease(&key, false, &mut executor)
+                .await?
+            {
+                ensure!(expire_time > current, NoLeaderSnafu);
+                Ok(leader.as_bytes().into())
+            } else {
+                NoLeaderSnafu.fail()
+            }
+        }
+    }
+
+    async fn resign(&self) -> Result<()> {
+        todo!()
+    }
+
+    fn subscribe_leader_change(&self) -> broadcast::Receiver<LeaderChangeMessage> {
+        self.leader_watcher.subscribe()
+    }
+}
+
+impl MySqlElection {
```

Comment:
It closely resembles `PgElection`, so we may use `RdsElection` to handle most of the election logic, but not in this PR, we can do this in the next PR.